tests/console/zypper_lr.pm

```diff
@@ -11,12 +11,102 @@
 use base "consoletest";
 use strict;
 use testapi;
+use utils;
+
+sub validatelr {
+    my ($args) = @_;
+
+    my $product         = uc $args->{product};
+    my $product_channel = $args->{product_channel} || "";
+    my $version         = get_var('VERSION');
+    my $uri             = $args->{uri};
+
+    assert_script_run "zypper lr -d | grep $product$version-*$product_channel.*$product$version-*$product_channel.*$uri";
+}
 
 sub run() {
+    # sets for addons
+    my (%h_addons, %h_addonurl, %h_scc_addons);
+    my @addons_keys     = split(/,/, get_var('ADDONS'));
+    my @addonurl_keys   = split(/,/, get_var('ADDONURL'));
+    my @scc_addons_keys = split(/,/, get_var('SCC_ADDONS'));
```

Comment:
These three will produce undef warnings when those vars are not set. Either use `get_required_var` if job should fail or `get_var('ADDONS', '')`.